src/SimpleSAML/Auth/Source.php

```diff
@@ -136,13 +136,13 @@ public function reauthenticate(array &$state): void
     /**
      * Complete authentication.
      *
-     * This function should be called if authentication has completed. It will never return,
+     * This function should be called if authentication has completed. It will return a Response,
      * except in the case of exceptions. Exceptions thrown from this page should not be caught,
      * but should instead be passed to the top-level exception handler.
      *
      * @param array &$state Information about the current authentication.
      */
-    public static function completeAuth(array &$state): void
+    public static function completeAuth(array &$state): Response
```

Comment:
Side-quest: should we change the signature to non-static & protected, since all calls to this are made by child-classes?